packages/utils/src/syncpromise.ts

```diff
@@ -218,24 +170,20 @@ class SyncPromise<T> implements PromiseLike<T> {
     this._handlers = [];
 
     cachedHandlers.forEach(handler => {
-      if (handler.done) {
+      if (handler[0]) {
         return;
       }
 
       if (this._state === States.RESOLVED) {
-        if (handler.onfulfilled) {
-          // eslint-disable-next-line @typescript-eslint/no-floating-promises
-          handler.onfulfilled((this._value as unknown) as any);
-        }
+        // eslint-disable-next-line @typescript-eslint/no-floating-promises
+        handler[1]((this._value as unknown) as any);
```

Comment:
Don’t we still need there if statements guard?

Reply:
i did not find a code path where this was ever not set. I changed the type declaration accordingly to require the handler.